Add cert troubleshooting tips (#3625)
zowe/docs-site: docs/troubleshoot/troubleshoot-zos-certificate.md

# Troubleshooting certificate configuration

You may encounter problems when configuring certificates. Review the following article for troubleshoot errors or warnings that can occur when configuring certificates.

## PKCS12 server keystore generation fails in Java 8 SR7FP15, SR7 FP16, and SR7 FP20

**Symptoms**

Some Zowe Desktop applications do not work when Zowe creates a PKCS12 keystore. A message may appear in the log such as the following:
* ZWES1060W Failed to init TLS environment, rc=1(Handle is not valid)
* ZWES1065E Failed to configure https server. Check agent https setting.

These messages indicate that ZSS cannot read the generated keystore. As such, parts of Zowe are not functional. 

**Solutions**

This error results from the incompatibility between Java and GSK regarding cryptography.

Try one of the following options if you are affected by this error:

- Temporarily downgrade Java, for example to Java 8 SR7FP10, and generate the PKCS12 keystore again.

> - Use the flags as presented in the following codeblock when generating a keystore: 

Reviewer: Where are these flags supposed to go? I think most users rely on the `zwe init certificates` which reads the `zowe.yaml` but I myself wouldn't know where to configure this.

Reviewer 2: I believe it could be used when you generating the keystore manually using java keytool, it is troubleshooting section so it is fine to have something that not covered by `zwe init certificates` here, it is kind of hotfix on customer side. 
Anyway it all comes from [this issue](https://github.com/zowe/docs-site/issues/2459) and there in discussion Andrej Ch. proposes to use shorter version `-J-Dkeystore.pkcs12.legacy` instead, so we can remove the confusing long flags and keep only legacy flag.

Dkeystore.pkcs12.keyPbeIterationCount=50000
```

- Set the flag `keystore.pkcs12.legacy` to enabled with no value to create a PKCS12 keystore that can be loaded.

:::note

> * If you already have an existing keystore or you are using keyrings, this error will not happen.

Reviewer: I think the existing keystore part may need to clarify that it needs to be existing but created with a compatible java version

Author: Does this capture the addition: "Ensure that the existing keystore part is  created with a compatible java version."

> * If you do not use ZSS, this error will not happen. ZSS is enabled by default.

Reviewer 2: This section looks useless, user comes to the troubleshooting section when something happened already, they do not need to know when this will not happen. However we can switch this to a useful information, in case the customer just sees error in the log, like "If you are not planning to use Zowe Desktop then you can disable ZSS component to get rid of this error in the log"

> * If you do not use ZSS, this error will not happen because ZSS is on by default.

Reviewer: ```suggestion
* If you do not use ZSS, this error will not happen. ZSS is enabled by default.
```
Something like this maybe? the because part sounds odd to me

> * If you already use your own PKCS12 files instead of the files that Zowe generates for you, this error will not happen. 

Reviewer: probably same clarification as the first item

:::

## Eureka request failed when using entrusted signed z/OSMF certificate

**Symptoms**

> When using the entrusted signed z/OSMF certificate an problem may occur whereby the AppServer cannot register with Eureka. Review of the logs indicate that the cause is the self-signed certificate:

Reviewer: ```suggestion
When using the entrusted signed z/OSMF certificate a problem may occur whereby the ZLUX AppServer cannot register with Eureka. Review of the logs indicate that the cause is the self-signed certificate:
```

```
<ZWED:198725> ZWESVUSR WARN (_zsf.bootstrap,webserver.js:156) ZWED0148E - Exception thrown when reading SAF keyring, e= Error: R_datalib call failed: function code: 01, SAF rc: 8, RACF rc: 8, RACF rsn: 44
```

> **Solutions**

Reviewer: ```suggestion
**Solution**
```

The error indicates that the keyring does not exist or cannot be found.

Look at the keyring information and confirm the corresponding certificate authorities. Ensure that you specify the `certificateAuthorities` variable with the correct keyring label and the label of the conected CA in the `zowe.certificate` section of your `zowe.yaml` file. 

For example, if the keyring label is `ZoweKeyring` and the LABLCERT of the connected CA is `CA Internal Cert`, the `certificateAuthorities` varaible should be `certificateAuthorities: safkeyring://ZWESVUSR/ZoweKeyring&CA Internal Cert`.

> ## Error reported: wrong certificate setup (keyring certificate)

Reviewer: Should the previous headline "Eureka request failed when using entrusted signed z/OSMF certificate" also have the "Error reported: " prefix?

Author: I'd like to add it but the draft where this is from did not include it and the contributor is no longer a Zowe contributor.

**Symptoms**

> The connection failed but the certificate appears to be correct. A keyring certificate is setup which does not need a value for `password` in the `zowe.certificate.keystore.password` and `zowe.certificate.truststore.password`. 

Reviewer 2: We could add more details here as well, like "The certificate appears to be correct, but the Gateway and the Discovery Service are failing to start" also it would be nice to add a real error code that is being shown, but i do not remember it

> **Solutions**

Reviewer: ```suggestion
**Solution**
```

> The password is only used for USS PKCS12 certificate files. The keyring is protected by SAF permissions. It is recommended to assign a value to `password` as shown in the following example:

Reviewer 2: The password is only used for USS PKCS12 certificate files. The keyring is protected by SAF permissions. 
_But some configurations can't work with empty password field,_ so it is recommended to assign a value to `password` as shown in the following example:

> The password is only used for USS PKCS12 certificate files. The keyring is protected by SAF permissions. It is recommended to fill the `password` with *password* as shown in the following example:

Reviewer: I think some starts fail if `password` is blank, maybe some more emphasis on the need to have it set?

**Example:**
```
certificate:
    keystore:
      type: JCERACFKS
      file: safkeyring:////ZWESVUSR/ZWEKeyring
      password: password
      alias: ZoweCert
    truststore:
      type: JCERACFKS

>       file: safkeyring:////ZWESVUSR/ZWEKering

Reviewer 2: A typo here: "ZWEKering" and below in certificateAuthorities field
